python/src/odin_data/client.py

```diff
@@ -1,113 +1,127 @@
 import argparse
-import time
-import datetime
-import threading
 import logging
 import sys
 import os
 import json
-import zmq
-from struct import Struct
 
 from odin_data.control.ipc_channel import IpcChannel
 from odin_data.control.ipc_message import IpcMessage
 
-
 try:
     from json.decoder import JSONDecodeError
 except ImportError:
-    JSONDecodeError = ValueError 
+    JSONDecodeError = ValueError
+
+class OdinDataClient(object):
 
-class FrameReceiverClient(object):
-    
     def __init__(self):
 
         prog_name = os.path.basename(sys.argv[0])
-        
+
         # Parse command line arguments
         self.args = self._parse_arguments(prog_name)
-        
+
         # create logger
         self.logger = logging.getLogger(prog_name)
         self.logger.setLevel(logging.DEBUG)
-        
+
         # create console handler and set level to debug
         ch = logging.StreamHandler(sys.stdout)
         ch.setLevel(logging.DEBUG)
-        
+
         # create formatter
         formatter = logging.Formatter('%(asctime)s %(levelname)s %(name)s - %(message)s')
-        
+
         # add formatter to ch
         ch.setFormatter(formatter)
-        
+
         # add ch to logger
         self.logger.addHandler(ch)
 
         # Create the appropriate IPC channels
         self.ctrl_channel = IpcChannel(IpcChannel.CHANNEL_TYPE_DEALER)
         self.ctrl_channel.connect(self.args.ctrl_endpoint)
-        
+
+        self._msg_id = 0
         self._run = True
-    
+
     def _parse_arguments(self, prog_name=sys.argv[0]):
-        
-        parser = argparse.ArgumentParser(prog=prog_name, description='ODIN Frame Receiver Client')
+
+        parser = argparse.ArgumentParser(prog=prog_name, description='odin-data application client')
         parser.add_argument('--ctrl', type=str, default='tcp://127.0.0.1:5000', dest='ctrl_endpoint',
                             help='Specify the IPC control channel endpoint URL')
         parser.add_argument('--config', type=argparse.FileType('r'), dest='config_file', nargs='?',
-                            default=None, const=sys.stdin, 
+                            default=None, const=sys.stdin,
                             help='Specify JSON configuration file to send as configure command')
+        parser.add_argument('--request_config', action='store_true',
```

Comment:
Could be have dashes?

```suggestion
        parser.add_argument('--request-config', action='store_true',
```